### PR TITLE
fix: handle tenant not found error

### PR DESCRIPTION
**Pull request by KShivendu**
## Summary of change

handle tenant not found error

## Related issues

-   https://github.com/supertokens/supertokens-node/pull/639

## Checklist for important updates

-   [ ] Changelog has been updated
-   [ ] `coreDriverInterfaceSupported.json` file has been updated (if needed)
    -   Along with the associated array in `supertokens_python/constants.py`
-   [ ] `frontendDriverInterfaceSupported.json` file has been updated (if needed)
-   [ ] Changes to the version if needed
    -   In `setup.py`
    -   In `supertokens_python/constants.py`
-   [ ] Had installed and ran the pre-commit hook
-   [ ] Issue this PR against the latest non released version branch.
    -   To know which one it is, run find the latest released tag (`git tag`) in the format `vX.Y.Z`, and then find the latest branch (`git branch --all`) whose `X.Y` is greater than the latest released tag.
    -   If no such branch exists, then create one from the latest released branch.
-   [ ] If have added a new web framework, update the `supertokens_python/utils.py` file to include that in the `FRAMEWORKS` variable
-   [ ] If added a new recipe that has a User type with extra info, then be sure to change the User type in supertokens_python/types.py
-   [ ] Make sure that `syncio` / `asyncio` functions are consistent.

### CODE REVIEW

File: supertokens_python/recipe/multitenancy/recipe.py [hunk 1]
```diff
@@ -216,6 +216,9 @@ async def login_methods_get(
             tenant_id, user_context
         )
```
Comment (rishabhpoddar) [inline supertokens_python/recipe/multitenancy/recipe.py:217]: why do you have two api imple for multi tenancy? One here, and one in supertokens_python/recipe/multitenancy/api/implementation.py

File: supertokens_python/recipe/thirdpartypasswordless/recipeimplementation/implementation.py [hunk 2]
```diff
@@ -15,7 +15,7 @@
 
 from typing import TYPE_CHECKING, Any, Dict, List, Optional, Union
 
-from supertokens_python.recipe.thirdparty.provider import ProviderInput
+from supertokens_python.recipe.thirdparty.provider import ProviderInput, Provider
```
Comment (rishabhpoddar) [inline supertokens_python/recipe/thirdpartypasswordless/recipeimplementation/implementation.py:18]: why has this been imported from here, but other places its imported from ThirdPartyInterfaces? Is this a mistake or deosn't matter?

**Review by rishabhpoddar — CHANGES_REQUESTED** [review]

File: supertokens_python/recipe/thirdpartypasswordless/recipeimplementation/implementation.py (hunk 2 above)
Comment (KShivendu) [reply supertokens_python/recipe/thirdpartypasswordless/recipeimplementation/implementation.py:18]: Doesn't matter.

File: supertokens_python/recipe/multitenancy/recipe.py (hunk 1 above)
Comment (KShivendu) [reply supertokens_python/recipe/multitenancy/recipe.py]: Removed the redundant one from `recipe.py`

**Pull request merged**